src/higher-level/lsm-hl.c

```diff
@@ -422,6 +423,7 @@ float unit_perim = 0;
         if (phl->lsm.oavg) lsm.avg_[f][p] = mx;
         if (phl->lsm.ogeo) lsm.geo_[f][p] = exp(logSum / (float)logCounter);
         if (phl->lsm.omax) lsm.max_[f][p] = maxVal;
+        if (phl->lsm.oare) lsm.are_[f][p] = validDataPixels;
```

Comment:
a couple of lines before, please add something like:

`if (validDataPixels > SHRT_MAX) validDataPixels = SHRT_MAX;`

probably use a new variable (`short` datatype) for this to make sure it is not re-used accidentally (e.g. in line 427)

Reply:
also, make it `private` in the OpenMP pragma